static/app/components/smartSearchBar/index.tsx

```diff
@@ -408,6 +410,9 @@ class SmartSearchBar extends Component<DefaultProps & Props, State> {
     ) {
       // eslint-disable-next-line react/no-did-update-set-state
       this.setState(this.makeQueryState(addSpace(query ?? undefined)));
+    } else if (disallowWildcard !== lastDisallowWildcard) {
+      // Re-parse query to apply new options (without resetting it to the query prop value)
+      this.setState(this.makeQueryState(this.state.query));
```

Comment:
I see that there's `// eslint-disable-next-line react/no-did-update-set-state` above the previous setState. Do we also need it here or can we remove it from there?

Reply:
right, that is not needed anymore 👍